Bottom: 5e43fd297dce9c2e2902cde4b8a817241016300e
Top:    fd4daefc01ea9e7294fc6a6fe37f199ef7bba470
Author: Jason Wang <jasowang@redhat.com>
Date:   2018-02-08 17:54:45 +0800

Refresh of vhost-packed-ring-support

---

diff --git a/drivers/vhost/vhost.c b/drivers/vhost/vhost.c
index bcac12b..5caaf50 100644
--- a/drivers/vhost/vhost.c
+++ b/drivers/vhost/vhost.c
@@ -2359,8 +2359,8 @@ static int vhost_add_used_n_packed(struct vhost_virtqueue *vq,
 		desc.len = heads[i].len;
 		set_desc_used(&desc, vq->used_warp_counter);
 
-		/* Update the flags before id and len */
-		smp_wmb();
+		/* Update flags etc before desc is written */
+		smp_mb();
 
 		used_idx = vq->last_used_idx & (vq->num - 1);
 		ret = vhost_copy_to_user(vq, vq->desc_packed + used_idx,
